code/game/objects/items/devices/uplinks.dm

```diff
@@ -29,9 +29,15 @@ A list of items and costs is stored under the datum of every game mode, alongsid
 		welcome = "THANKS FOR MAPPING IN THIS THING AND NOT CHECKING FOR RUNTIMES BUDDY"
 		uses = 90 // Because this is only happening on centcomm's snowflake uplink
 
-/obj/item/device/uplink/proc/refund(mob/user, obj/item/I)
+/obj/item/device/uplink/proc/refund(mob/living/carbon/human/user, obj/item/I)
 	if(!user || !I)
 		return
+	if (istype(I, /obj/item/stack/sheet/mineral/telecrystal/refined))
+		var/obj/item/stack/sheet/S = I
+		uses += S.amount
+		user.drop_item(S, src)
+		to_chat(user, "<span class='notice'>You insert [S.amount] telecrystal inside the uplink.</span>")
```

Comment:
telecrystal may be plural if amount isn't 1

also "into" would be better than "inside"